lightly/api/api_workflow_tags.py

```diff
@@ -0,0 +1,77 @@
+from typing import *
+
+from lightly.api.bitmask import BitMask
+from lightly.openapi_generated.swagger_client import TagData, \
+    TagArithmeticsRequest, TagArithmeticsOperation, TagBitMaskResponse
+
+
+class _TagsMixin:
+
+    def _get_all_tags(self) -> List[TagData]:
+        """ Gets all tags on the server
+
+        Returns:
+
+        """
+        return self._tags_api.get_tags_by_dataset_id(self.dataset_id)
+
+    def get_tag_and_filenames(
```

Comment:
call this `get_tag_with_filenames`

Reply:
and I would just remove the argument "filenames_on_server" and simplify the workflow

Reply:
actually, I would split the function into three:

```python
def get_tag_by_id(self, tag_id: str):
    pass

def get_tag_by_name(self, tag_name: str):
    # convert tag_name to tag_id here
    return self.get_tag_by_id(tag_id)

def get_filenames_in_tag(self, tag):
    pass
```

Reply:
If you do it this way you can also keep the optional `filenames_on_server` argument because then the function isn't such a beast